R/jira.R

```diff
@@ -8,103 +8,392 @@
+          "16x16" = comment_detail$author_avatarUrls_16x16,
+          "32x32" = comment_detail$author_avatarUrls_32x32
+        ),
+        displayName = comment_detail$author_displayName,
+        active = comment_detail$author_active,
+        timeZone = comment_detail$author_timeZone
+      ),
+      body = comment_detail$body,
+      updateAuthor = list(
+        self = comment_detail$updateAuthor_self,
+        name = comment_detail$updateAuthor_name,
+        key = comment_detail$updateAuthor_key,
+        avatarUrls = list(
+          "48x48" = comment_detail$updateAuthor_avatarUrls_48x48,
+          "24x24" = comment_detail$updateAuthor_avatarUrls_24x24,
+          "16x16" = comment_detail$updateAuthor_avatarUrls_16x16,
+          "32x32" = comment_detail$updateAuthor_avatarUrls_32x32
+        ),
+        displayName = comment_detail$updateAuthor_displayName,
+        active = comment_detail$updateAuthor_active,
+        timeZone = comment_detail$updateAuthor_timeZone
+      ),
+      created = comment_detail$created,
+      updated = comment_detail$updated
+    )
+    comments[[length(comments) + 1]] <- comment
+  }
+  return(list(comments = comments))
+}
+
+# details of comments to be added
+comment_details <- list(
+  list(
+    self = "https://example.com/jira/comment/1",
+    id = "1",
+    author_self = "https://example.com/jira/user?username=user1",
+    author_name = "user1",
+    author_key = "user1",
+    author_avatarUrls_48x48 = "https://example.com/avatar/user1_48x48.jpg",
+    author_avatarUrls_24x24 = "https://example.com/avatar/user1_24x24.jpg",
+    author_avatarUrls_16x16 = "https://example.com/avatar/user1_16x16.jpg",
+    author_avatarUrls_32x32 = "https://example.com/avatar/user1_32x32.jpg",
+    author_displayName = "User One",
+    author_active = TRUE,
+    author_timeZone = "Etc/UTC",
+    body = "First comment body.",
+    updateAuthor_self = "https://example.com/jira/user?username=user2",
+    updateAuthor_name = "user2",
+    updateAuthor_key = "user2",
+    updateAuthor_avatarUrls_48x48 = "https://example.com/avatar/user2_48x48.jpg",
+    updateAuthor_avatarUrls_24x24 = "https://example.com/avatar/user2_24x24.jpg",
+    updateAuthor_avatarUrls_16x16 = "https://example.com/avatar/user2_16x16.jpg",
+    updateAuthor_avatarUrls_32x32 = "https://example.com/avatar/user2_32x32.jpg",
+    updateAuthor_displayName = "User Two",
+    updateAuthor_active = TRUE,
+    updateAuthor_timeZone = "America/New_York",
+    created = "2021-01-01T10:00:00.000+0000",
+    updated = "2021-01-01T12:00:00.000+0000"
+  )
+
+)
+
+#' Create Jira Sample Log Issues with comments
+#'
+#' Creates a sample jira issue with two components WITH comments
+#'
+#' @param folder_path An optional path to where the sample .json should be created.
+#' @return The path to the sample .json file.
+#' @export
+#' @family {unittest}
+jira_create_sample_log_issues_comments <- function(folder_path="/tmp") {
```

Comment:
This should be (not in this form) as an example in example.R. Remember the jira_create_sample_log will go away, much as the git_sample_log. Think how the git_create_sample_log became example functions.